File: services/ai/app/schemas/health.py
"""Health response schemas.

Mirrors the `HealthReport` TypeScript interface in packages/shared so the
NestJS health module can consume this service's report without translation.
"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field

HealthStatus = Literal["ok", "degraded", "down"]


class DependencyHealth(BaseModel):
    name: str
    status: HealthStatus
    latency_ms: float | None = Field(default=None, serialization_alias="latencyMs")
    message: str | None = None

    model_config = {"populate_by_name": True}


class HealthReport(BaseModel):
    status: HealthStatus
    service: str
    version: str
    environment: str
    uptime_seconds: float = Field(serialization_alias="uptimeSeconds")
    timestamp: str
    dependencies: list[DependencyHealth] = Field(default_factory=list)

    model_config = {"populate_by_name": True}

File: packages/shared/src/types/health.ts
/** Health contract shared by the NestJS API and the Python AI service. */

export const HEALTH_STATUSES = ['ok', 'degraded', 'down'] as const;

export type HealthStatus = (typeof HEALTH_STATUSES)[number];

export interface DependencyHealth {
  /** Dependency name, e.g. `database`, `redis`, `aiService`. */
  name: string;
  status: HealthStatus;
  /** Round-trip time of the probe in milliseconds. */
  latencyMs: number | null;
  /** Populated when the probe failed or reported a degraded state. */
  message?: string;
}

export interface HealthReport {
  /** Worst status across all dependencies. */
  status: HealthStatus;
  service: string;
  version: string;
  environment: string;
  /** Process uptime in seconds. */
  uptimeSeconds: number;
  timestamp: string;
  dependencies: DependencyHealth[];
}
